tornado/web.py

```diff
@@ -2278,7 +2278,11 @@ def initialize(self, url, permanent=True):
         self._permanent = permanent
 
     def get(self, *args):
-        self.redirect(self._url.format(*args), permanent=self._permanent)
+        to_url = self._url.format(*args)
+        if self.request.query:
+            to_url = httputil.url_concat(
+                to_url, urlparse.parse_qsl(self.request.query))
```

Comment:
If `url_concat` passed `doseq=True` to `urlencode`, then I think the `parse_qs`-like dict would work. Is that a safe change to make? I'm not sure why you'd want the `doseq=False` behavior but maybe people are relying on it.

If we don't want to change `url_concat`, you could unwind `query_arguments` like this: 

    def qs_to_qsl(qs):
        for k, vs in qs.items():
            for v in vs:
                yield (k, v)
    qsl = list(qs_to_qsl(self.request.query_arguments))

Or as a one-liner, `sum((list(zip([k]*len(vs), vs)) for (k, vs) in qs.items()), [])`. But I prefer the less-cryptic generator function.